app/services/xccdf_report_parser.rb

```diff
@@ -51,21 +58,65 @@ def check_for_external_reports
     # rubocop:disable Style/GuardClause
     if external_report?
       raise ExternalReportError,
-            "No policy found matching benchmark #{benchmark.ref_id} "\
-            "(RHEL-#{benchmark.os_major_version}) and profile "\
-            "#{test_result_profile.ref_id} with host #{@host.name} assigned "\
-            "for account #{@account.account_number}."
+            "No policy found matching benchmark #{benchmark.ref_id}. "\
+            "#{parse_failure_message}"
     end
     # rubocop:enable Style/GuardClause
   end
 
+  def check_for_missing_benchmark
+    # rubocop:disable Style/GuardClause
+    unless benchmark.persisted?
+      raise UnknownBenchmarkError,
+            "No benchmark found matching ref_id #{benchmark.ref_id} and "\
+            "SSG #{benchmark.version}. #{parse_failure_message}"
+    end
+    # rubocop:enable Style/GuardClause
+  end
+
+  def check_for_missing_test_result_profile
+    # rubocop:disable Style/GuardClause
+    unless test_result_profile.persisted?
```

Comment:
Related to `test_result_profile`:
* There seems to be a bug within the second lookup
```
::Profile.find_or_initialize_by(
  ref_id: @test_result_file.test_result.profile_id,
  name: @test_result_file.test_result.profile_id,
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  benchmark_id: benchmark.id
)
```
* there could be a memoization
* the method is private (which probably explains the absent memoization)

Reply:
- Wow! How long has that been there? lol It's definitely incorrect and probably should not include `name` at all. I'll submit a separate PR with that change to make sure nothing breaks.
- Any memoization here is OK and beneficial
- The XccdfReportParser included the Xccdf::Hosts module, so it has access to all private methods included in it. Why use private at all, then? Well, it's mostly just trying to communicate intention at that point, similar to Python's "private" methods (`_foo`). I guess if we're going to use it outside the module, we should make the method public.

Reply:
@vkrizan ok, so I did some more investigation of `test_result_profile`, and this code doesn't cause any unintended side effects. This test_result_profile is only ever used during cloning to create a profile in the user's account. As you can see, the method attempts to find the profile by benchmark and profile ref_id, and then creates a new object if not found. We can essentially replace `find_or_initialize_by` with `new`. It's kind of weird, for sure, but it isn't causing any issues.

Reply:
Updated in a separate commit.